cmd/pebble/cmd_enter.go

```diff
@@ -0,0 +1,186 @@
+given), and the subcommand is run as a pebble client command that can
+communicate with the service manager instance in the background. The service
+manager is shut down after the commands finishes unless noted otherwise below.
+
+Only a subset of pebble commands is supported as a subcommand:
+
+  exec
+  help
+  ls
+  plan
+  services
+  start
+  version
+
+Furthermore, some commands have restrictions and idiosyncrasies:
+
+  exec, ls, plan, services, version
+    Service manager logging is disabled unless -v/--verbose flag is provided.
+
+  help, ls, plan, services, version
+    Service autostart (the --run flag) is prohobitied.
+
+  start, stop
+    Service manager continues running after the command finishes.
+
+  stop
+    Service autostart (the --run flag) is required.
+`
+
+type enterFlags int
+
+const (
+	enterFlagSilenceLogging enterFlags = 1 << iota
+	enterFlagNoServiceManager
+	enterFlagKeepServiceManager
+	enterFlagRequireServiceAutostart
+	enterFlagProhibitServiceAutostart
+)
+
+type cmdEnter struct {
+	clientMixin
+
+	sharedRunEnterOpts
+	Run        bool `long:"run"`
+	Positional struct {
+		Cmd []string `positional-arg-name:"<subcommand>"`
+	} `positional-args:"yes"`
+}
+
+func init() {
+	optsHelp := map[string]string{
+		"run": "Start \"autostart\" services before executing subcommand",
+	}
+	for k, v := range sharedRunEnterOptsHelp {
+		optsHelp[k] = v
+	}
+	cmdInfo := addCommand("enter", shortEnterHelp, longEnterHelp, func() flags.Commander { return &cmdEnter{} }, optsHelp, nil)
+	cmdInfo.extra = func(cmd *flags.Command) {
+		cmd.PassAfterNonOption = true
+	}
+}
+
+func commandEnterFlags(commander flags.Commander) (enterFlags enterFlags, supported bool) {
+	supported = true
+	switch commander.(type) {
+	case *cmdExec:
+		enterFlags = enterFlagSilenceLogging
+	case *cmdHelp:
+		enterFlags = enterFlagNoServiceManager
+	case *cmdLs, *cmdPlan, *cmdServices, *cmdVersion:
+		enterFlags = enterFlagSilenceLogging | enterFlagProhibitServiceAutostart
+	case *cmdStart:
+		enterFlags = enterFlagKeepServiceManager
+	case *cmdStop:
+		enterFlags = enterFlagKeepServiceManager | enterFlagRequireServiceAutostart
+	default:
+		supported = false
+	}
+	return
+}
+
+func (cmd *cmdEnter) Execute(args []string) error {
+	if len(args) > 0 {
+		return ErrExtraArgs
+	}
+
+	runCmd := cmdRun{
+		sharedRunEnterOpts: cmd.sharedRunEnterOpts,
+	}
+	runCmd.setClient(cmd.client)
+
+	if len(cmd.Positional.Cmd) == 0 {
+		runCmd.run(nil)
+		return nil
+	}
+
+	runCmd.Hold = !cmd.Run
+
+	var (
+		commander flags.Commander
+		extraArgs []string
+	)
+
+	parser := Parser(cmd.client)
+	parser.CommandHandler = func(c flags.Commander, a []string) error {
+		commander = c
+		extraArgs = a
+		return nil
+	}
+
+	if _, err := parser.ParseArgs(cmd.Positional.Cmd); err != nil {
+		return err
+	}
+
+	enterFlags, supported := commandEnterFlags(commander)
+
+	if !supported {
+		return fmt.Errorf("enter: this command is not supported")
```

Comment:
`enter: subcommand %q is not supported`

Otherwise when we get the error while running a script or something we don't know what is the actual issue.

It also seems better to call these as "subcommands" since "enter" is the command here.